common/src/provider.rs

```diff
@@ -1,18 +1,48 @@
 //! Commonly used helpers to construct `Provider`s
 
 use crate::{ALCHEMY_FREE_TIER_CUPS, REQUEST_TIMEOUT};
+use async_trait::async_trait;
 use ethers_core::types::{Chain, U256};
 use ethers_middleware::gas_oracle::{GasCategory, GasOracle, Polygon};
 use ethers_providers::{
-    is_local_endpoint, Http, HttpRateLimitRetryPolicy, Middleware, Provider, RetryClient,
-    RetryClientBuilder, DEFAULT_LOCAL_POLL_INTERVAL,
+    is_local_endpoint, Http, HttpRateLimitRetryPolicy, JsonRpcClient, LooseHttp, Middleware,
+    Provider, RetryClient, RetryClientBuilder, DEFAULT_LOCAL_POLL_INTERVAL,
 };
 use eyre::WrapErr;
 use reqwest::{IntoUrl, Url};
-use std::{borrow::Cow, time::Duration};
+use serde::{de::DeserializeOwned, Serialize};
+use std::{borrow::Cow, fmt::Debug, time::Duration};
+
+/// Wraps strict or loose rpc handling in the Http provider
+#[derive(Debug)]
+pub enum ProviderTypes {
```

Comment:
would like to rename this
```suggestion
pub enum ProviderKind {
```